codegen/src/main/kotlin/software/amazon/smithy/rust/codegen/smithy/generators/UnionGenerator.kt

```diff
@@ -47,18 +47,18 @@ class UnionGenerator(
         writer.rustBlock("impl ${unionSymbol.name}") {
             sortedMembers.forEach { member ->
                 val memberSymbol = symbolProvider.toSymbol(member)
-                val funcNamePart = member.memberName.toSnakeCase()
-                val variantName = member.memberName.toPascalCase()
+                val variantName = symbolProvider.toMemberName(member).toPascalCase()
+                val funcNamePart = variantName.toSnakeCase()
 
                 if (sortedMembers.size == 1) {
                     Attribute.Custom("allow(irrefutable_let_patterns)").render(this)
                 }
-                rust("/// Tries to convert the enum instance into its #D variant.", unionSymbol)
-                rust("/// Returns `Err(&Self) if it can't be converted.` ")
+                rust("/// Tries to convert the enum instance into [`$variantName`](#T::$variantName).", unionSymbol)
```

Comment:
This looks incorrect to me? In the screenshot, it should have been `bool` instead of `Bool`.

Reply:
hmm that's a good point since the API actually returns the inner type. I linked to the outer type because that's where the docs are